src/ansi.rs

```diff
@@ -715,6 +718,73 @@ bitflags! {
     }
 }
 
+/// Construct escape sequences for regular keys, like `1`, when
+/// modified by `Shift-`, `Control-`, `Alt-` or `Meta-`modifiers.
+/// This doesn't apply to the named keys on the keyboard, like `F1`.
+/// The escape sequence also doesn't apply to defined unicode input, like
+/// cyrillic chars.
+///
+/// For the details on how to implement the mode handling correctly
+/// consult the xterm's [`implementation`] and the output of the
+/// xterm's provided perl [`script`].
+///
+/// [`implementation`]: https://invisible-island.net/xterm/modified-keys.html
+/// [`script`]: https://github.com/ThomasDickey/xterm-snapshots/blob/master/vttests/modify-keys.pl
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ModifyOtherKeys {
+    /// Reset the state.
+    Reset,
+    /// Enables this feature for keys except for those with well-
+    /// known behavior, e.g., Tab, Backarrow and some special
+    /// control character cases which are built into the X11
+    /// library, e.g., Control-Space to make a NUL, or Control-3
+    /// to make an Escape character.
```

Comment:
```suggestion
    /// Enables this feature except for keys with well-
    /// known behavior, e.g., Tab, Backspace and some special
    /// control character cases which are built into the X11
    /// library (e.g. Control-Space to make a NUL, or Control-3
    /// to make an Escape character).
```

Reply:
I used the xterm docs here, if it matters.

Reply:
Not the worst I suppose, I still prefer my suggestions. :P